### PR TITLE
MINOR: Use JUnit 5 in raft module

### PR DESCRIPTION
**Pull request by ijuma**
I also removed a test class with no tests currently.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: raft/src/test/java/org/apache/kafka/raft/VotedStateTest.java [hunk 1]
```diff
@@ -1,28 +0,0 @@
...
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.apache.kafka.raft;
-
-import org.junit.Test;
-
-public class VotedStateTest {
```
Comment (hachikuji) [inline raft/src/test/java/org/apache/kafka/raft/VotedStateTest.java:21]: Ouch. Probably meant to come back to this. Filed https://issues.apache.org/jira/browse/KAFKA-10519.

**Review by hachikuji — APPROVED** [review]
Thanks @ijuma . LGTM

File: raft/src/test/java/org/apache/kafka/raft/VotedStateTest.java (hunk 1 above)
Comment (ijuma) [reply raft/src/test/java/org/apache/kafka/raft/VotedStateTest.java:21]: Thanks, you saved me from having to do that. :)

**Comment by ijuma** [comment]
Unrelated flaky test failures:

```
Build / JDK 11 / kafka.api.ConsumerBounceTest.testRollingBrokerRestartsWithSmallerMaxGroupSizeConfigDisruptsBigGroup
Build / JDK 15 / kafka.api.ConsumerBounceTest.testRollingBrokerRestartsWithSmallerMaxGroupSizeConfigDisruptsBigGroup
Build / JDK 15 / kafka.api.ConsumerBounceTest.testRollingBrokerRestartsWithSmallerMaxGroupSizeConfigDisruptsBigGroup
Build / JDK 15 / org.apache.kafka.streams.integration.EosBetaUpgradeIntegrationTest.shouldUpgradeFromEosAlphaToEosBeta[true]

```

**Pull request merged**